src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
Move the ternary operator inside the filter method as I wrote above

Just replace p with a better name

And just think about the destruction of the element 

Something like this `people.filter(({ sex, name }) => ...` or 
```
people.filter(person => {
 const { sex, name} = person;
}
```


And make this logic in another function

Just read about destruction assignment [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#description)